strict-version-matcher-plugin/src/main/java/com/google/android/gms/dependencies/DataObjects.kt

```diff
@@ -102,16 +102,16 @@ data class Dependency(val fromArtifactVersion: ArtifactVersion, val toArtifact:
 class ArtifactDependencyManager {
   /** Synchronized access to the dependencies to ensure elements aren't added while it's being iterated to provide a copy via the getter. */
   private val dependencyLock = Object()
-  @VisibleForTesting internal val dependencies: HashMap<Artifact, ArrayList<Dependency>> = HashMap()
+  @VisibleForTesting internal val dependencies: HashMap<Artifact, HashSet<Dependency>> = HashMap()
 
   fun addDependency(dependency: Dependency) {
     synchronized(dependencyLock) {
-      var depListForArtifact = dependencies.get(dependency.toArtifact)
-      if (depListForArtifact == null) {
-        depListForArtifact = ArrayList()
-        dependencies[dependency.toArtifact] = depListForArtifact
+      var depSetForArtifact = dependencies.get(dependency.toArtifact)
+      if (depSetForArtifact == null) {
+        depSetForArtifact = HashSet()
```

Comment:
nit: Should this be LinkedHashSet to ensure determinism?

Reply:
I don't think that's needed here because the data is copied into a HashSet before being used in the calling class.

Reply:
Ah, I see `getDependencies` now. Thanks for checking.